src/connection.rs

```diff
@@ -108,11 +108,17 @@ impl Connection {
 
     //Assumes that there will never be nested arrays in a redis response.
     async fn parse_array(start: &[u8], mut stream: &mut TcpStream) -> Result<Value> {
-        let num = String::from_utf8_lossy(&start[1..])
+        let num_parsed = String::from_utf8_lossy(&start[1..])
             .trim()
-            .parse::<usize>()
+            .parse::<i32>()
             .unwrap();
 
+        // result can be negative (blpop/brpop return '-1' on timeout)
+        if num_parsed < 0 {
+            return Ok(Value::Array(Vec::new()));
```

Comment:
The Redis protocol specification recommends returning a nil object rather than an empty array for null arrays. I think it may be the best option here as well, considering they are the same thing. It would also make the code easier to read down the line.
```suggestion
            return Ok(Value::Nil);
```